### ISSUE TITLE
Home -> NewProblemsForm:  Missing Accessible Feedback on Success or Failure

### ISSUE THREAD

**Issue by johnhbenetech** [benetech/MathShare#771]
When the “Add Problem” button has been activated, the application attempts to validate and save the current editor input.  If that process is successful, the new problem is added to the current set and appears in the table at the top of the dialog, but no accessible feedback is given to a screen reader user to indicate that anything has happened.  Similarly, if the input is invalid, nothing seems to happen either. 
 
Suggested Solution 
 
Provide some sort of informative, accessible feedback when the “Add Problem” button is pressed, which informs the user of either a successful or failed outcome.  It could be as simple as announcing a string of text via the user’s screen reader such as, “Added new problem in position N”, or, “Problem input is invalid”.  Alternatively, if an error has occurred, a level 2 heading could be rendered somewhere in the dialog, followed by some explanatory text to explain what went wrong.  The heading could then receive keyboard focus automatically (meaning it would need a tabindex attribute with a value of “-1”).

Aha! Link: https://diagramlabs.aha.io/features/B-126

**Comment by rupeshparab** [comment]
@jscholes 
The new alert library was added to support custom html in alerts, the reason we needed it was to add tabIndex to the heading in the alert.

This is the markup of the custom alert:
https://github.com/benetech/MathShare/commit/077291249e3d6cfc6195ec7ec2cfbff41ee34ce8#diff-952f8edd178813f5f634edb7c2c0ba35R15

**Comment by jscholes** [comment]
@rupeshparab sure, I understand.  I'm just saying that I don't hear an alert being fired when I use this modal to add a problem.

**Comment by rupeshparab** [comment]
@jscholes 
I looked into it, the addition of the library is actually related to this commit: https://github.com/benetech/MathShare/commit/3ca81d0fc2f1efc97756158ec97032b3f4c92505

The current issue was fixed by using aria live:
https://github.com/benetech/MathShare/commit/077291249e3d6cfc6195ec7ec2cfbff41ee34ce8#diff-d6c554c49cb8ebaf1fc1fc515c3ab7f5R129

I tried to test it now, it seems to be malfunctioning on VoiceOver too. I can see the dom changing in aria-live but it gets read only when I close the add problem modal. I am investigating if anything on the modal is blocking the aria-live announcement.

**Comment by rupeshparab** [comment]
I have identified the issue, it was due to a conflict with aria modal library which was silencing aria-live, I have pushed a fix for this.

**Comment by jscholes** [comment]
Fixed in https://github.com/benetech/MathShare/commit/f32b3c411ec8cff2728d5d1ba4513128887cd909.
